app/services/actions/prospective_student/upsert.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module Actions
+  module ProspectiveStudent
+    class Upsert
+      attr_reader :params
+
+      def initialize(params, academic_term: AcademicTerm.current, type: :bulk)
+        @params = params.merge(
+          system_register_type: type,
+          academic_term_id:     academic_term&.id,
+          expiry_date:          academic_term&.end_of_term
+        )
+      end
+
+      class << self
+        def call(params, academic_term: AcademicTerm.current, type: :bulk)
+          new(params, academic_term: academic_term, type: type).call
+        end
+      end
+
+      def call
+        prepare_params
```

Comment:
Parametrelerin her `call`da reset'lenmesi gerekiyor mu? Callback'i initialization sonrasına taşımda side-effect yaratır mı?

Reply:
Bu kullanımda aslında resetleme filan olmuyor çünkü tek kullanımlık işini yapan servisler hatta `new` metodunu private metod yapmakta fayda var. Diğer konuya gelirsek initialize ile burada olması  arasında bu servis için bir fark yok tabi bunu yapmamın amacı kodu okuyanın sadece `call` metoduna odaklanması ve tüm iş akışını görebilmesi.  `initialize`'a alsakta `side-effect` olmaz.

Reply:
> hatta new metodunu private metod yapmakta fayda var. 

:+1: